language/bytecode-verifier/src/signature.rs

```diff
@@ -199,25 +199,11 @@ impl<'a> SignatureChecker<'a> {
                                         self.module.struct_handle_at(struct_def.struct_handle);
                                     let type_actuals =
                                         &self.module.locals_signature_at(*type_actuals_idx).0;
-                                    let mut errs = check_generic_instance(
+                                    check_generic_instance(
                                         context,
                                         &struct_handle.type_formals,
                                         type_actuals,
-                                    );
```

Comment:
anything else we need to do here for the verifier?

Reply:
Don't know the verifier code well enough to say for sure, but I think it would be useful to test corner cases around "inferred resources":

```
module M {
struct MaybeResource<Type> { t: Type }

public yes1(v: LibraCoin.T) {
  move_to_sender<MaybeResource<LibraCoin.T>>(MaybeResource { t: v });
}

public yes2<R: resource>(r: R) {
  move_to_sender<MaybeResource<R>>(MaybeResource { t: r });
}

public no1(v: u64) {
  move_to_sender<MaybeResource<u64>>(MaybeResource { t: v });
}

public no2<A>(a: A) {
  move_to_sender<MaybeResource<A>>(MaybeResource { t: a });
}

public no3<C: copyable>(c: C) {
  move_to_sender<MaybeResource<C>>(MaybeResource { t: c });
}
}
```

Reply:
it turns out those are all invalid examples because we can only publish declared resources and not inferred resources

Reply:
Ah, good catch! This seems like a very sensible policy to me, but it would interesting to explore whether it would be sound to allow examples like `yes1` and `yes2` (just for our own understanding, not because I think we need this feature).